gprofiler/spark/spark_collector.py

```diff
@@ -0,0 +1,894 @@
+                name_property = config_property.find("name")
+                if (
+                    name_property is not None
+                    and name_property.text is not None
+                    and name_property.text.startswith("yarn.resourcemanager.webapp.address")
+                ):
+                    value_property = config_property.find("value")
+                    if value_property is not None and value_property.text is not None:
+                        return value_property.text
+        if self._master_address is not None:
+            return self._master_address
+        else:
+            host_name = self._get_yarn_host_name(resource_manager_process)
+            return host_name + ":8088"
+
+    def _guess_mesos_master_webapp_address(self, process: psutil.Process) -> str:
+        if self._master_address:
+            return self._master_address
+        else:
+            host_name = get_hostname()
+            return host_name + ":5050"
+
+    def _get_yarn_host_name(self, resource_manager_process: psutil.Process) -> str:
+        host_name = self._get_yarn_config_property(resource_manager_process, "yarn.resourcemanager.hostname")
+        return host_name if host_name is not None else get_hostname()
+
+    def _is_yarn_master_collector(self, resource_manager_process: psutil.Process) -> bool:
+        """
+        yarn lists the addresses of the other masters in order communicate with
+        other masters, so we can choose one of them (like rm1) and run the
+        collection only on him so we won't get the same metrics for the cluster
+        multiple times the rm1 hostname is in both EMR and Azure us the internal
+        dns and it's starts with the host name for examplem in EMR:
+        rm1 = 'ip-10-79-63-183.us-east-2.compute.internal:8025' where the host
+        name is 'ip-10-79-63-183' for example in
+        azure: 'rm1 = hn0-nrt-hb.3e3rqto3nr5evmsjbqz0pkrj4g.tx.internal.cloudapp.net:8050'
+        where the host name is 'hn0-nrt-hb.3e3rqto3nr5evmsjbqz0pkrj4g'
+        """
+        rm1_address = self._get_yarn_config_property(resource_manager_process, "yarn.resourcemanager.address.rm1", None)
+        host_name = self._get_yarn_host_name(resource_manager_process)
+
+        if rm1_address is None:
+            self._logger.info(
+                "yarn.resourcemanager.address.rm1 is not defined in config, so it's a single master deployment,\
+                     enabling spark collector..."
+            )
+            return True
+
+        is_collection_master = rm1_address.startswith(host_name)
+        if is_collection_master:
+            self._logger.info(
+                f"this is the collector master, because rm1: {rm1_address} \
+                    starts with the host name: {host_name}, enabling spark collector..."
+            )
+        else:
+            self._logger.info(
+                f"this is not the collector master, because rm1: {rm1_address}\
```

Comment:
```suggestion
                f"This is not the collector master, because rm1: {rm1_address!r}\
```
Also please remove the `\`